libraries/AP_HAL/Device.h

```diff
@@ -36,7 +36,8 @@ class AP_HAL::Device {
         SPEED_LOW,
     };
 
-    typedef void PeriodicHandle;
+    FUNCTOR_TYPEDEF(PeriodicCb, bool);
+    typedef void* PeriodicHandle;
```

Comment:
By being opaque you give more freedom to the HAL implementations on how to do it. I think that after we implement all of them, we find everyone is doing the same thing, we can change them to be an object rather than on opaque pointer.